### ISSUE TITLE
ML: Address class imbalance in OrfGroupClassifier (LightGBM) — add scale_pos_weight and retrain

### ISSUE THREAD

**Issue by roeimed0** [roeimed0/bacterial-gene-prediction#111]
## Problem

`OrfGroupClassifier` uses LightGBM to predict whether a group of nested ORFs contains a real gene. The training set has a ~10–20:1 negative:positive ratio at the group level (derived from ~44:1 at the ORF level), but `scale_pos_weight` is never set. The threshold was manually lowered to 0.1 (from 0.5) post-hoc, indicating the imbalance was noticed but never properly fixed.

## What exists already

- `extract_group_features()` — full feature engineering pipeline, ready to use for training
- Trained model stored as `models/orf_classifier_lgb.pkl` (joblib format)
- Feature names stored alongside as `models/feature_names.pkl`

### 1. Add a `train()` method to `OrfGroupClassifier`

```python
def train(self, X: pd.DataFrame, y: np.ndarray, feature_names: list):
    n_pos = y.sum()
    n_neg = len(y) - n_pos
    self.model = lgb.LGBMClassifier(
        n_estimators=300,
        learning_rate=0.05,
        scale_pos_weight=n_neg / n_pos,
        ...
    )
    self.model.fit(X, y)
    self.feature_names = feature_names
```

### 2. Add PR-curve threshold selection

After training, sweep thresholds on a held-out validation set and pick the one maximising F1 instead of using a hardcoded 0.1.

### 3. Add `scripts/train_lgb.py`

Driver script that:
1. Runs the scoring pipeline on a set of GENOME_CATALOG genomes (training split)
2. Labels groups as positive (overlaps reference GFF) or negative
3. Calls `OrfGroupClassifier.train()`
4. Evaluates on a held-out validation split
5. Saves retrained model to `models/`

## Expected impact

Recall improvement without precision loss. The model currently over-predicts negatives; proper class weighting should shift the learned decision boundary.

## Related

- Companion issue: HybridGeneFilter focal loss / pos_weight retraining
- Supersedes #98

**Comment by roeimed0** [comment]
Implemented and validated. OrfGroupClassifier retrained with scale_pos_weight on 100 GENOME_CATALOG genomes. Group-level class ratio was 1.1:1 (nearly balanced), so the main gain came from retraining on more data with early stopping. Threshold calibrated to 0.07 via PR curve. +2.3pp F1, +3.1pp precision on 20 held-out genomes at same recall.
